### ISSUE TITLE
Translations are missing the namespace prefix of Apps from Managed Packages

### ISSUE THREAD

**Issue by amtrack** [forcedotcom/cli#2905]
### Summary

When retrieving `Translations` from a Source-Tracked Org, the `<customApplications>` entries are missing the namespacePrefix of Apps from Managed Packages (e.g. [Address Picker Autocomplete for Flow](https://appexchange.salesforce.com/appxListingDetail?listingId=a0N3A00000FKAgjUAH)).

As a consequence the deployment of those invalid `.translation-meta.xml` files to another org fails with:

`In field: name - no CustomApplication named Address_Picker_Autocomplete found`

### Steps To Reproduce

Please see my Git repository for all details and a reproduction using GitHub Actions:
https://github.com/mdapi-issues/mre-translations-missing-namespace-of-applications

TL;DR

```console
# create a Scratch Org with the Translation Workbench enabled
sf org create scratch -f config/project-scratch-def.json --alias translation-managed-application --set-default
# install the Managed Package "Address Picker Autocomplete for Flow"
sf package install --package 04t6F000004DJCEQA4 --no-prompt --wait 30
# deploy some unrelated translation of a managed CustomLabel
sf project deploy start --source-dir fixtures/some-unrelated-translation-of-a-managed-customlabel
# retrieve the translations from the Scratch Org
sf project retrieve start -m Translations
```

### Expected result

```xml
<?xml version="1.0" encoding="UTF-8"?>
<Translations xmlns="http://soap.sforce.com/2006/04/metadata" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
    <customApplications>
        <description xsi:nil="true"/>
        <label><!-- Address Picker Autocomplete --></label>
        <name>DV_Flow_AP__Address_Picker_Autocomplete</name>
    </customApplications>
</Translations>
```

### Actual result

```xml
<?xml version="1.0" encoding="UTF-8"?>
<Translations xmlns="http://soap.sforce.com/2006/04/metadata" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
    <customApplications>
        <description xsi:nil="true"/>
        <label><!-- Address Picker Autocomplete --></label>
        <name>Address_Picker_Autocomplete</name>
    </customApplications>
</Translations>
```

actual (red) vs. expected (green)

```diff
<?xml version="1.0" encoding="UTF-8"?>
<Translations xmlns="http://soap.sforce.com/2006/04/metadata" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
    <customApplications>
        <description xsi:nil="true"/>
        <label><!-- Address Picker Autocomplete --></label>
-        <name>Address_Picker_Autocomplete</name>
+        <name>DV_Flow_AP__Address_Picker_Autocomplete</name>
    </customApplications>
</Translations>
```

### System Information

```json
{
  "architecture": "darwin-arm64",
  "cliVersion": "@salesforce/cli/2.42.6",
  "nodeVersion": "node-v20.10.0",
  "osVersion": "Darwin 23.5.0",
  "rootPath": "/Users/john.doe/.local/lib/node_modules/@salesforce/cli",
  "shell": "zsh",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 3.0.18 (core)",
    "@oclif/plugin-commands 3.3.4 (core)",
    "@oclif/plugin-help 6.0.22 (core)",
    "@oclif/plugin-not-found 3.1.9 (core)",
    "@oclif/plugin-plugins 5.0.21 (core)",
    "@oclif/plugin-search 1.0.24 (core)",
    "@oclif/plugin-update 4.2.12 (core)",
    "@oclif/plugin-version 2.1.2 (core)",
    "@oclif/plugin-warn-if-update-available 3.0.19 (core)",
    "@oclif/plugin-which 3.1.10 (core)",
    "@salesforce/cli 2.42.6 (core)",
    "apex 3.1.11 (core)",
    "auth 3.6.11 (core)",
    "data 3.3.5 (core)",
    "deploy-retrieve 3.6.14 (core)",
    "info 3.2.8 (core)",
    "limits 3.3.7 (core)",
    "marketplace 1.2.5 (core)",
    "org 4.1.10 (core)",
    "packaging 2.4.4 (core)",
    "schema 3.3.6 (core)",
    "settings 2.2.6 (core)",
    "sobject 1.4.2 (core)",
    "source 3.3.12 (core)",
    "telemetry 3.3.7 (core)",
    "templates 56.2.7 (core)",
    "trust 3.6.11 (core)",
    "user 3.5.8 (core)"
  ]
}
```

### Additional information

![image](https://github.com/forcedotcom/cli/assets/789765/6f558af1-a5ff-4fa0-8f4f-bd0775328b64)

![Untitled](https://github.com/forcedotcom/cli/assets/789765/932c6479-dda4-41be-a28a-18c03b508aaa)

![Screenshot 2024-06-05 at 16 00 29](https://github.com/forcedotcom/cli/assets/789765/e76d3013-896b-4e0e-a829-deefa3a0f9ed)

**Comment by amtrack** [comment]
Sorry for spamming you with issues related to the Metadata API. 😄

**Comment by mshanemc** [comment]
I'll add this to our list of items to discuss with them

**Comment by shetzel** [comment]
Possibly related to this one (if you dig deep in that thread and take note of CustomObjectTranslation issues): https://github.com/forcedotcom/cli/issues/2759

As always with server side issues, it will greatly help get their attention with a customer support case.

**Comment by amtrack** [comment]
Thanks @mshanemc @shetzel. I'll create a customer support case.

**Comment by frukeci** [comment]
Hi @amtrack, currently facing the same issue - is this still being investigated or did the support come up with a solution?

**Comment by amtrack** [comment]
@frukeci I was told by Salesforce Support on the phone that a Known Issue will be created.

**Comment by amtrack** [comment]
@frukeci Here's the Known Issue: https://issues.salesforce.com/issue/a028c00000zJBW9AAO. Please click "Report" to let Salesforce know this affects you.

**Comment by frukeci** [comment]
@amtrack Thank you! I reported it.

**Comment by gearsetdave** [comment]
We're seeing this for translations of a namespaced layout too, so I think the issue is wider than the known issue. I've clicked Report anyway 👍

**Comment by axmav** [comment]
Same problem for us, we will very appreciate the fix.

**Comment by amtrack** [comment]
Apparently this has been resolved. I cannot reproduce it anymore. Thanks! 👍
